### PR TITLE
More sensors support: NTX US and undocumented tilt for EV3 Gyro

### PR DESCRIPTION
**Pull request by malinnikov**
Also removed some code duplications.

### CODE REVIEW

**Comment by malinnikov** [comment]
Video with the Gyro tilt mode tested: https://youtu.be/WfKWzBpmrdQ

File: API/ev3_sensor.c [hunk 1]
```diff
@@ -245,34 +251,26 @@ void* ReadSensorData(int sensorPort)
 			return readNewDumbSensor(sensorPort);
 			// Lightsensor
 		case COL_REFLECT:
-			return readUartSensor(sensorPort);
```
Comment (malinnikov) [inline API/ev3_sensor.c]: These repeated readUartSensor calls are removed to leave the only one call for all relates cases.

**Comment by a3f** [comment]
Changes look good to me. Do you agree to relicense your changes under the MPL-2.0?
Also sorry for the merge conflict, If you like, you can force push a rebase onto current master, otherwise I'll do it on the eventual merge.

Sorry for the delay!

**Comment by malinnikov** [comment]
I rebased onto your master, thanks for the review.

And I agree to relicense the changes in the most convenient way, so

I, Pavel Malinnikov, give permission to license my contributions to the EV3-API under any version of Mozilla Public License.

**Comment by a3f** [comment]
Ah, the sensor codes has been under the terms of the GPL-3.0-or-later anyway, so no need to worry about the MPL there. I'll merge your changes under the terms of the GPL-3.0-or-later. Thanks again.

**Comment by a3f** [comment]
I'll see if there are objection to relicense under the GPL-2.0-or-later. Do you agree to relicense your contributions under the GPL-2.0-or-later?

**Comment by malinnikov** [comment]
Sure. I, Pavel Malinnikov, give permission to license my contributions to the EV3-API under any version of public License.

**Pull request merged**